test/libsolidity/SemanticTest.cpp

```diff
@@ -124,6 +146,15 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 	return result;
 }
 
+vector<string> SemanticTest::effectsOfCall(FunctionCall const& _call) const
+{
+	vector<string> effects;
+	for (auto const& hook: m_hooks)
+		for (auto const& effect: hook(_call))
+			effects.emplace_back(effect);
+	return effects;
+}
```

Comment:
Is it sufficient for call effects to always return a string?

Reply:
Yes, I think so. We could convert everything that we need int this context into a string.